mock/src/mock/account.rs

```diff
@@ -149,42 +149,64 @@ pub fn mock_account(nonce: Felt, code: Option<AccountCode>, assembler: &Assemble
     Account::new(account_id, account_vault, account_storage, account_code, nonce)
 }
 
-pub fn mock_fungible_faucet(account_id: u64, assembler: &Assembler) -> Account {
+pub fn mock_fungible_faucet(
+    account_id: u64,
+    nonce: Felt,
+    empty_reserved_slot: bool,
+    assembler: &Assembler,
+) -> Account {
+    let initial_balance = if empty_reserved_slot {
+        ZERO
+    } else {
+        Felt::new(FUNGIBLE_FAUCET_INITIAL_BALANCE)
+    };
```

Comment:
Question: could we infer whether to set the initial balance or not based on the nonce? For example:
```Rust
let initial_balance = if nonce == ZERO {
    ZERO
} else {
    Felt::new(FUNGIBLE_FAUCET_INITIAL_BALANCE)
};
```

Reply:
We want to build invalid mock accounts such that we can test the failure cases of the protocol.  I.e when the faucet account is created (`nonce == ZERO`) with a non-empty reserved slot.